src/main/java/com/basho/riak/client/api/commands/kv/MultiCommand.java

```diff
@@ -54,9 +54,8 @@ protected MultiCommand(Builder builder)
 
         Submitter submitter = new Submitter(operations, maxInFlight, cluster, future);
 
-        Thread t = new Thread(submitter);
-        t.setDaemon(true);
-        t.start();
+        new Thread(submitter).start();
+
```

Comment:
So the only thing the new thread is used for is kicking off the individual scalar commands, one at a time until it reaches it's max concurrency.  If we're interrupted while doing that we set `fail` on the multicommand future, and let the thread die.  This thread shouldn't have ownership over anything so letting it die is clean and humane. 

https://github.com/basho/riak-java-client/blob/17aa8296a7b642bd5b53737e37dcae28b54601d1/src/main/java/com/basho/riak/client/api/commands/kv/MultiCommand.java#L253-L271

Once it reaches the end of the `run()` block, [the JVM will kill the thread](http://stackoverflow.com/questions/4283270/how-do-i-cleanup-a-thread-after-its-run-method-finishes), so no `join()` is needed.